app/src/main/java/ch/epfl/sdp/offline/EventSaver.java

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.sdp.offline;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import ch.epfl.sdp.Event;
+import ch.epfl.sdp.db.DatabaseObject;
+
+public class EventSaver extends ObjectSaver<Event> {
+
+    @Override
+    protected String getCollectionString() {
+        return "Events";
+    }
+
+    /**
+     * Saves Event in a file
+     * @param toSave The object to save in the collection
+     * @param docReference Id of the document (Event)
+     * @param deleteDate When we can delete the temp file
+     */
+    public void saveEvent(Event toSave, String docReference, Date deleteDate, File path) throws IOException, ClassNotFoundException {
+        HashMap<String, Map<String,Object>> statusFiles = getEventStatusFiles(path);
+
+        Map<String,Object> metaData = new HashMap<String, Object>();
+        Date lastModifyDate = new Date();
+        metaData.put("deleteDate",deleteDate);
+        metaData.put("lastModifyDate",lastModifyDate);
+
+        saveFile(toSave, docReference, path);
+
+        //update status
+        statusFiles.put(docReference, metaData);
+        updateEventStatusFiles(statusFiles,path);
+
+        //remove old files
+        for (String key: statusFiles.keySet()) {
+            Date deleteDateStorage = (Date) statusFiles.get(key).get("deleteDate");
+            if (deleteDateStorage.before(new Date())){
+                removeSingleEvent(key, path);
+            }
+        }
+    }
+
+    public List<Event> getAllEvents(File path) throws IOException, ClassNotFoundException {
+        HashMap<String, Map<String,Object>> statusFiles = getEventStatusFiles(path);
+        List<String> listReference = new ArrayList<>(statusFiles.keySet());
```

Comment:
remove first line and directly call `getEventStatusFiles(path).keySet()`

Reply:
je trouve que c'est plus clair comme ca